src/view-resources.js

```diff
@@ -67,6 +73,72 @@ interface ViewEngineHooks {
 * Represents a collection of resources used during the compilation of a view.
 */
 export class ViewResources {
+
+  static convention(target: Function) {
+    let config = target.resource;
+    let resource;
+    if (config) {
+      if (typeof config === 'string') {
+        // it's a custom element, with name is the resource variable
+        // static resource = 'my-element'
+        resource = new HtmlBehaviorResource();
+        resource.elementName = config;
+      } else {
+        if (typeof config === 'function') {
+          // static resource() {  }
+          config = config();
+        }
+        if (typeof config === 'string') {
+          // static resource() { return 'my-custom-element-name' }
+          config = { name: config };
+        }
+        // after normalization, copy to another obj
+        config = Object.assign({}, config);
```

Comment:
Nitpick: could be a `else` from previous checks, avoids one allocation and one copy.

Also: this could crash. For example if I do `resource: 4` in my code or any type that's neither a string, an object or a function returning `string | object`. `resource() { return null }` would equally crash, etc.

Reply:
That would be too passive, probably should be job for documentation, or I think we can throw an error to let devs know ?

Reply:
We could throw a custom error so devs could know easier what going on in case.

Reply:
Yes. 
A random unexplained crash is bad. An error that says _"view of {target.name} should be a string or an object"_ is a lot more helpful.